### DISCUSSION TITLE
Slide Picker

### DISCUSSION THREAD

**Discussion by coreapostrophe** [pmndrs/react-spring#1728] [Support]
## 💬About
I'm trying to do a slide picker similar to those you'll find in modern **mobile alarms UI**. I can imagine a straight-forward way of creating this by creating a list of horizontally aligned numbers, attaching a react-use-gesture drag animation to its Y position, and then using the width of each element to check which number is currently in focus based on the scroll amount.

However, with that method, I'll have to be dependent on the pixel width and height of each item.  I'm wondering if there's a better method of achieving this?

## 👀 Preview
**Note:** I don't need it to be 3D! I just want to do the sliding picker with edge items decreasing in opacity as they are moved out of the viewable area.
![image](https://user-images.githubusercontent.com/6149731/139396966-2e689484-7651-4c88-89e0-96b4c89b6e5e.png)

> Thank you for reading!

**Comment by joshuaellis** [comment]
I guess a starting point to understand the opacity is the distance from the center of the container? Imagine if you drew an imaginary line on the x axis of the middle, say it's 300px from 0 (0 being the top of your container), you can then change the opacity based on the difference from 300.

I don't think this is a react-spring specific implementation, _but_ you could use `useSprings` and when using the `start` method of the imperative API, use the index to find it's position and set it's opacity like that?
